heron/tools/cli/src/python/execute.py

```diff
@@ -101,8 +101,15 @@ def heron_tar(class_name, topology_tar, arguments, tmpdir_root, java_defines):
   '''
   # Extract tar to a tmp folder.
   tmpdir = tempfile.mkdtemp(dir=tmpdir_root, prefix='tmp')
+  tmpdir_abs = os.path.abspath(tmpdir)
 
   with contextlib.closing(tarfile.open(topology_tar)) as tar:
+    # Validate file paths
+    for name in tar.getnames():
+      untarpath_abs = os.path.abspath(os.path.join(tmpdir, name))
+      if not untarpath_abs.startswith(tmpdir_abs):
```

Comment:
hmm. In which case that `os.path.abspath(os.path.join(tmpdir, name))` result does not start with `tmpdir_abs`?

Reply:
@nwangtw Is your question implying that this change might not be necessary?

Reply:
@nwangtw Following up on my prior question. Should we proceed with this PR? Are there any changes you would like to see?

Reply:
@nicknezis It might be necessary. I wasn't sure about the exact cases that trigger the issue and it is not easy to reason the code. Maybe add a comment with the vulnerability link so that in the future it is easy for other people to understand the code.

Reply:
I updated the comment to include a the vulnerability link. I'll merge this pull request.